jmclient/jmclient/client_protocol.py

```diff
@@ -301,6 +301,22 @@ def on_JM_TX_RECEIVED(self, nick, txhex, offer):
             self.defaultCallbacks(d)
         return {"accepted": True}
 
+    @commands.JMTXBroadcast.responder
+    def on_JM_TX_BROADCAST(self, txhex):
+        """ Makers have no issue broadcasting anything,
+        so only need to prevent crashes.
+        Note in particular we don't check the return value,
+        since the transaction being accepted or not is not
+        our (maker)'s concern.
+        """
+        try:
+            txbin = hextobin(txhex)
+            jm_single().bc_interface.pushtx(txbin)
+        except:
+            jlog.info("We received an invalid transaction broadcast "
+                      "request: " + txhex)
```

Comment:
`pushtx()` has try .. except inside it, should never throw. Return value (True/False) should be checked instead.

Reply:
But `Error` exception should be handled as it will be thrown by `binascii.unhexlify()` if non-hex characters are passed to it.

Reply:
OK, a couple of things going on here. First, do note the comment above that specifically says why we don't check the return value of  `pushtx`.

On the other hand, pushtx does include a `bintohex` call outside the try block, which can throw `TypeError` if it isn't passed bytes.

I'd still rather catch everything in these cases - this is a task which is not something the Maker cares about, what he cares about is that the bot does not crash under any external input. It's more a practical decision than a technically ideal one. Do you agree?

Reply:
Yes, you're right, only hex conversions should throw there currently, push errors will be logged as warnings. And if that changes it's better to have try block and not 100% correct error message in that case will not be a disaster.